pkg/kubelet/kubelet.go

```diff
@@ -1722,12 +1752,41 @@ func (kl *Kubelet) makePodDataDirs(pod *api.Pod) error {
 	return nil
 }
 
+// syncPod is the transaction script for the sync of a single pod.
+//
+// Arguments:
+//
+// pod - the pod to sync
+// mirrorPod - the mirror pod for the pod to sync, if it is a static pod
+// podStatus - the current status (TODO: always from the status manager?)
+// updateType - the type of update (CREATE,ADD,DELETE,RECONCILE)
+//
+// The workflow is:
+//  1.  If the pod is being created, record pod worker start latency
+//  2.  Call generateAPIPodStatus to prepare an api.PodStatus for the pod
+//  3.  If the pod is being seen as running for the first time, record pod
+//      start latency
+//  4.  Update the status of the pod in the status manager
+//  5.  Kill the pod if it should not be running
+//  6.  Create a mirror pod if the pod is a static pod, and does not
+//      already have a mirror pod
+//  7.  Create the data directories for the pod if they do not exist
+//  8.  Mount volumes and update the volume manager
+//  9.  Fetch the pull secrets for the pod
+// 10.  Call the container runtime's SyncPod callback
+// 11.  Update the traffic shaping for the pod's ingress and egress limits
+//
+// If any step if this workflow errors, the error is returned, and is repeated
+// on the next syncPod call.
 func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecontainer.PodStatus, updateType kubetypes.SyncPodType) error {
+	// (0) Latency measurements for the main workflow are relative to the
+	// (first time the pod was seen by the API server.
 	var firstSeenTime time.Time
 	if firstSeenTimeStr, ok := pod.Annotations[kubetypes.ConfigFirstSeenAnnotationKey]; ok {
 		firstSeenTime = kubetypes.ConvertToTimestamp(firstSeenTimeStr).Get()
 	}
 
+	// (1) Record pod worker start latency if being created
```

Comment:
i see that you try to relate (1) with 1. above in workflow description, but numbers will change, so I think we should just remove them.